examples/forum/example_server.py

```diff
@@ -0,0 +1,114 @@
+import socket
+import sys
+import socketserver
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
+import urllib.parse
+import forum_util as ut
+
+# using port numbers prepended with 9 to avoid calling sudo during test
+servport = 9090
+posts_file = "./posts/posts.txt"
+
+
+def getCommentsFromUser(fd, usr):
+    output_txt = "["
+    comment = fd.read()
+    cnt = 0
+    offset = 0
+    while True:
+        cmt_json = ut.read_simple_json(comment, offset)
+        if(cmt_json == None):
+            break
+        offset += (cmt_json['_offset'] + cmt_json['_length']) # relative positional seek
+        if(usr == None or cmt_json['user'] == usr):
+            cnt += 1
+            output_txt += (ut.jsonify_urllib_params(cmt_json) + ",")
+    print("Number of matches:", cnt)
+    output_txt = output_txt[0:-1] + "]" # replace newline at the end of last comment
+    if(cnt == 0):
+        return None
+    return output_txt
+
+# https://wiki.python.org/moin/BaseHttpServer
+class MyHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        print("")
+        print("GET for", self.client_address)
+        print("Path:", self.path)
+
+        output_txt = ""
+        data_dict = {}
+        try:
+            # params is everything after "?"
+            params = self.path[self.path.index("?")+1:]
+            data_dict = urllib.parse.parse_qs(params)
+        finally:
+            print("Request params:", data_dict)
+
+        try:
+            fd = open(posts_file, "r")
+            # If no user specified, get all comments.
+            if not 'user' in data_dict:
+                output_txt = getCommentsFromUser(fd, None)
+            # Fetch comments from specified user
+            else:
+                output_txt = getCommentsFromUser(fd, data_dict['user'][0])
+                if(output_txt == None):
+                    output_txt = "This user has no comments."
+            fd.close()
+        except:
+            output_txt = ("\nError opening file \"" + posts_file + "\" or retrieving data.\n")
+            output_txt += "Did you make sure it exists? Maybe post something first.\n"
+            print(output_txt)
```

Comment:
I didn't mention this in my previous review, but normally if there's an error while processing a request we'll want to send a 500 (Server Error) response. I didn't check, but I think the client might only be processing 200 responses. This is fine for our case, but just more of an FYI